### PR TITLE
Add methods to facilitate threading

### PR DESCRIPTION
**Pull request by AuraWind**
Due to the memoization of Http::Client instances in a HalClient instance, sending concurrent requests through the same HalClient causes thread-safety issues

### CODE REVIEW

File: lib/hal_client/representation.rb [hunk 1]
```diff
@@ -297,8 +305,8 @@ def raw
       @raw
     end
 
-    # Returns the HalClient used to retrieve this representation
-    attr_reader :hal_client
+    # Set/Return the HalClient used to retrieve this representation
+    attr_accessor :hal_client
```
Comment (pezra) [inline lib/hal_client/representation.rb]: the hal_client ivar shouldn't be settable from public interface.

**Comment by AuraWind** [comment]
I've made the changes as requested.  Let me know if there's anything else.

File: lib/hal_client.rb [hunk 2]
```diff
@@ -5,6 +5,11 @@
 
 # Adapter used to access resources.
 class HalClient
+  # Operations on a HalClient instance are not thread-safe.  If you'd like to
+  # use a HalClient instance in a threaded environment, consider using the
+  # method #clone_for_use_in_different_thread to create a copy for each new
+  # thread
```
Comment (pezra) [inline lib/hal_client.rb]: please move this into the class docs (above the class def)

**Comment by AuraWind** [comment]
👍 Got it

**Comment by pezra** [comment]
released as v4.1.0. Thanks!

**Pull request merged**